### ISSUE TITLE
Creating custom ITokenFilter is ignoring configuration properties

### ISSUE THREAD

**Issue by Tasteful** [elastic/elasticsearch-net#5055]
**NEST/Elasticsearch.Net version**: 7.5.1

**Elasticsearch version**: 7.9.1

**Description of the problem including expected versus actual behavior**:
When I creating a custom `ITokenFilter` the properties on the filter is not included. The custom token filter is needed to be able to use an [plugin](https://github.com/Tasteful/elasticsearch-analysis-dynamic-synonym/) that is installed on the server to dynamic fetch the synonyms from external source instead of the internal file on the server.

**Steps to reproduce**:
1. Create a user defined token filter class.
 ```cs
public class DynamicSynonymTokenFilter : ITokenFilter
{
    public bool? Expand { get; set; }
    public SynonymFormat? Format { get; set; }
    public bool? Lenient { get; set; }
    public IEnumerable<string> Synonyms { get; set; }
    public string SynonymsPath { get; set; }
    public string Tokenizer { get; set; }
    public bool? Updateable { get; set; }
    public string Type { get; } = "dynamic_synonym";
    public string Version { get; set; }
    public int Interval { get; set; }
}
```

2. Configure the index to use the `UserDefined` token filter
```cs
        AnalysisDescriptor ConfigureIndex(AnalysisDescriptor analysisDescriptor, CultureInfo cultureInfo)
        {
            var filter = new DynamicSynonymTokenFilter
            {
                SynonymsPath = "https://my-synonym-server-url-that-not-is-relevant",
                Updateable = true,
                Lenient = true,
                Interval = 60
            };

            return analysisDescriptor
                .TokenFilters(filterDescriptor => filterDescriptor.UserDefined("synonymFilter", filter));
        }
```

3. Create the index

**Expected behavior**
The properties on the `DynamicSynonymTokenFilter`-class should be included in the generated JSON. Today the JSON is generated as below that missing multiple properties

Index creation request
```json
{
  "settings": {
    "analysis": {
      "filter": {
        "synonymFilter": {
          "type": "dynamic_synonym"
        }
      }
    }
  }
}
```
Index creation response
```json
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "dynamic synonym requires `synonyms_path` to be configured"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "dynamic synonym requires `synonyms_path` to be configured"
  },
  "status": 400
}
```

**Comment by Mpdreamz** [comment]
Thanks for reporting @Tasteful this is similar to this PR. Linking for reference: https://github.com/elastic/elasticsearch-net/pull/5039

**Comment by Mpdreamz** [comment]
I can not reproduce this on later versions of the client, this should be fixed as of 7.6.0 through 

https://github.com/elastic/elasticsearch-net/pull/5062

Thanks for reporting non the less!
